pkg/component/component.go

```diff
@@ -1462,19 +1529,47 @@ func getStorageFromConfig(localConfig *config.LocalConfigInfo) storage.StorageLi
 	return storageList
 }
 
+// GetComponentName gets the component name from current directory name
+func GetComponentName() (string, error) {
```

Comment:
we already have a similar function at https://github.com/openshift/odo/blob/a8a5eca39b80abf693bbe1e5e98880f9f31bf34a/pkg/odo/cli/component/devfile.go#L87.